src/main/java/com/feedzai/commons/sql/abstraction/dml/result/ResultIterator.java

```diff
@@ -126,6 +132,15 @@ public ResultIterator(PreparedStatement statement) throws DatabaseEngineExceptio
         this(statement, null, true);
     }
 
+    /**
+     * Retrieves the value for the index of the row being currently processed by the iterator.
+     *
+     * @return The value for the index of the row being currently processed by the iterator.
```

Comment:
- Add a note that if the ResultIterator is closed, then this will return 0.
- Also, make it clear that by current row index, you mean the index of the next row that will be returned by the next()/nextResult() method.

Reply:
Since index was sort of misleading, I changed "current row index" to "current row counter", and updated the documentation. I also added a unit test for this.